### PR TITLE
ci: add codeql config

### PR DESCRIPTION
**Pull request by corneliusroemer**
Adds CodeQL config, which automatically scans for security vulnerabilities.

Python and Typescript work without custom config, but kotlin required some.

### CODE REVIEW

**Comment by anna-parker** [comment]
@corneliusroemer I see this error when I look into the security section on github: 
image - will this PR fix this or is https://github.com/github/codeql-action/issues/2378#issuecomment-2327705067 potentially still causing issues?

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
```
Comment (theosanderson) [inline .github/workflows/codeql.yml:11]:
```suggestion
# For most projects, this workflow file will not need changing; you simply need
# to commit it to your repository.
#
# You may wish to alter this file to override the set of languages analyzed,
# or to provide custom queries or build logic.
#
# ******** NOTE ********
# We have attempted to detect the languages in your repository. Please check
# the `language` matrix defined below to confirm you have the correct set of
# supported CodeQL languages.
#
```

**Review by theosanderson — APPROVED** [review]
Seems useful!

**Comment by corneliusroemer** [comment]
Great question @anna-parker - I should have been explicit.

1. I set up codeql with default settings (no config file)
2. This ran, it worked for Python and TS, it failed for kotlin
3. This is the notification you see, that something failed
4. I then switched to advanced config (having explicit config file)
5. This is the PR here
6. Based on how codeql works, the advanced config doesn't make the warning disappear until it's merged into main
7. The codeql config here ran on this branch and it worked, so it shouldn't error again.

Hope this makes sense!

**Pull request merged**